tools/test-proxy/Azure.Sdk.Tools.TestProxy/Matchers/BodilessMatcher.cs

```diff
@@ -7,6 +7,11 @@ public class BodilessMatcher : RecordMatcher
         /// <summary>
         /// This matcher adjusts the "match" operation to EXCLUDE the body when matching a request to a recording's entries.
         /// </summary>
-        public BodilessMatcher() : base(false) { }
+        public BodilessMatcher() : base(false)
+        {
+            // when we are ignoring bodies, we should also ignore any header that is directly 
+            // influenced by said body
+            ExcludeHeaders.Add("Content-Length");
```

Comment:
We should probably look at updating RecordMatcher to reflect the changes in https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core.TestFramework/src/RecordMatcher.cs

In particular, we don't use ExcludeHeaders anymore for this and instead use IgnoredHeaders, which has the property that the presence of the header is still respected while matching. ExcludeHeaders doesn't care about the presence of the header, so it is more permissive.

Reply:
I'll take a look at this later today.